apps/web/src/scenes/TokenDetailPage/TokenDetailAsset.tsx

```diff
@@ -95,24 +113,40 @@ function TokenDetailAsset({ tokenRef, hasExtraPaddingForNote }: Props) {
     >
       <NftFailureBoundary tokenId={token.dbid}>
         <VisibilityContainer>
-          <AssetContainer isVisible={hasPreviewUrl}>
-            <StyledImage
-              src={cachedUrls[tokenId]?.url}
-              onLoad={handleNftLoaded}
-              height={resultDimensions.height}
-              width={resultDimensions.width}
-            />
-          </AssetContainer>
-          {shouldShowShimmer && (
-            <ShimmerContainer>
-              <Shimmer />
-            </ShimmerContainer>
+          {!showLightbox && (
+            <StyledLightboxButton onClick={toggleLightbox}>
+              <SearchIcon />
+            </StyledLightboxButton>
           )}
-          <AssetContainer isVisible={hasRawUrl}>
-            <NftDetailAssetComponent onLoad={handleRawLoad} tokenRef={token} />
-          </AssetContainer>
+          {lightboxContainer &&
+            createPortal(
+              <>
+                <AssetContainer isVisible={hasPreviewUrl}>
+                  <StyledImage
+                    src={cachedUrls[tokenId]?.url}
+                    onLoad={handleNftLoaded}
+                    height={resultDimensions.height}
+                    width={resultDimensions.width}
+                  />
+                </AssetContainer>
+                {shouldShowShimmer && (
+                  <ShimmerContainer>
+                    <Shimmer />
+                  </ShimmerContainer>
+                )}
+                <AssetContainer isVisible={hasRawUrl}>
+                  <NftDetailAssetComponent onLoad={handleRawLoad} tokenRef={token} />
+                </AssetContainer>
+              </>,
+              lightboxContainer
+            )}
```

Comment:
is this portal any different than the one created in `NftDetailLightbox.tsx`? maybe we can make it its own component?

Reply:
good idea, consolidated into a new component `NftDetailAssetContainer`